satellite/gracefulexit/endpoint_test.go

```diff
@@ -378,13 +378,135 @@ func TestInvalidStorageNodeSignature(t *testing.T) {
+		require.NoError(t, err)
+
+		// Process endpoint should return immediately if node is disqualified
+		response, err := processClient.Recv()
+		require.True(t, errs2.IsRPC(err, rpcstatus.PermissionDenied))
+		require.Nil(t, response)
+
+		// check that the exit has completed and we have the correct transferred/failed values
+		progress, err := satellite.DB.GracefulExit().GetProgress(ctx, exitingNode.ID())
+		require.NoError(t, err)
+
+		require.Equal(t, int64(0), progress.PiecesTransferred)
+		require.Equal(t, int64(1), progress.PiecesFailed)
+	})
+
+}
+
+func TestExitDisqualifiedNodeFailEventually(t *testing.T) {
+	testTransfers(t, numObjects, func(ctx *testcontext.Context, nodeFullIDs map[storj.NodeID]*identity.FullIdentity, satellite *testplanet.SatelliteSystem, processClient exitProcessClient, exitingNode *storagenode.Peer, numPieces int) {
+		disqualifyNode(t, ctx, satellite, exitingNode.ID())
+
+		deletedCount := 0
+		for {
+			response, err := processClient.Recv()
+			if errs.Is(err, io.EOF) {
+				// Done
+				break
+			}
+			if deletedCount >= numPieces {
+				// when a disqualified node has finished transfer all pieces, it should receive an error
+				require.True(t, errs2.IsRPC(err, rpcstatus.PermissionDenied))
+				break
+			} else {
+				require.NoError(t, err)
+			}
+
+			switch m := response.GetMessage().(type) {
+			case *pb.SatelliteMessage_TransferPiece:
+				require.NotNil(t, m)
+
+				pieceReader, err := exitingNode.Storage2.Store.Reader(ctx, satellite.ID(), m.TransferPiece.OriginalPieceId)
+				require.NoError(t, err)
+
+				header, err := pieceReader.GetPieceHeader()
+				require.NoError(t, err)
+
+				orderLimit := header.OrderLimit
+				originalPieceHash := &pb.PieceHash{
+					PieceId:   orderLimit.PieceId,
+					Hash:      header.GetHash(),
+					PieceSize: pieceReader.Size(),
+					Timestamp: header.GetCreationTime(),
+					Signature: header.GetSignature(),
+				}
+
+				newPieceHash := &pb.PieceHash{
+					PieceId:   m.TransferPiece.AddressedOrderLimit.Limit.PieceId,
+					Hash:      originalPieceHash.Hash,
+					PieceSize: originalPieceHash.PieceSize,
+					Timestamp: time.Now(),
+				}
+
+				receivingNodeID := nodeFullIDs[m.TransferPiece.AddressedOrderLimit.Limit.StorageNodeId]
+				require.NotNil(t, receivingNodeID)
+				signer := signing.SignerFromFullIdentity(receivingNodeID)
+
+				signedNewPieceHash, err := signing.SignPieceHash(ctx, signer, newPieceHash)
+				require.NoError(t, err)
+
+				success := &pb.StorageNodeMessage{
+					Message: &pb.StorageNodeMessage_Succeeded{
+						Succeeded: &pb.TransferSucceeded{
+							OriginalPieceId:      m.TransferPiece.OriginalPieceId,
+							OriginalPieceHash:    originalPieceHash,
+							OriginalOrderLimit:   &orderLimit,
+							ReplacementPieceHash: signedNewPieceHash,
+						},
+					},
+				}
+				err = processClient.Send(success)
+				require.NoError(t, err)
+			case *pb.SatelliteMessage_DeletePiece:
+				deletedCount++
+			case *pb.SatelliteMessage_ExitFailed:
```

Comment:
Will the node ever receive an official exit failed message if it is disqualified? If so, this is fine. Otherwise, I think we should remove this case since it could mislead the person reading the test into thinking it will happen.